src/emc/usr_intf/gmoccapy/gmoccapy.py

```diff
@@ -5783,12 +5735,58 @@ def _make_hal_pins(self):
 # =========================================================
 
 if __name__ == "__main__":
+    # Set up the base logger
+    #   We have do do this before importing other modules because on import
+    #   they set up their own loggers as children of the base logger.
+
+    # If log_file is none, logger.py will attempt to find the log file specified in
+    # INI [DISPLAY] LOG_FILE, failing that it will log to $HOME/<base_log_name>.log
+
+    # Note: In all other modules it is best to use the `__name__` attribute
+    #   to ensure we get a logger with the correct hierarchy.
+    #   Ex: LOG = logger.getLogger(__name__)
+
+    from qtvcp import logger
+    LOG = logger.initBaseLogger('GMoccapy', log_file=None, log_level=logger.WARNING)
+
+    # we set the log level early so the imported modules get the right level
+    if '-d' in sys.argv:
+        # Log level defaults to INFO, so set lower if in debug mode
+        logger.setGlobalLevel(logger.DEBUG)
+        LOG.debug('DEBUGGING logging on')
+    elif '-i' in sys.argv:
+        # Log level defaults to INFO, so set lower if in debug mode
```

Comment:
```suggestion
        # Log level defaults to WARNING, so set lower if in debug mode
```
defaults to WARNING if I get it right?

What about a flag for CRITICAL?

Reply:
yes i se the comments are wrong - copied from qtvcp and i changed the default level recently.
I'll need to fix them on qtvcp too. thanks

my opinion:
critical - in my mind if you have critical errors you should pretty much be logging then shutting down.
Having a critical log setting would block errors from showing. can't think of any reason you would want to hide errors.